### PR TITLE
#204: Handle displaying listings

### PR DESCRIPTION
**Pull request by dmutrokot**

### CODE REVIEW

File: src/api/thegraph.api.js [hunk 1]
```diff
@@ -230,6 +232,18 @@ export default {
         }).catch((error) => console.log(error));
     },
 
+    async getErc721ListingsBySeller(seller) {
+        return await this.getData(erc721ListingsBySeller(seller))
+            .then(response => response.data.erc721Listings)
```
Comment (Vladislav94) [inline src/api/thegraph.api.js:237]: please move callback to the newline

File: src/api/thegraph.api.js [hunk 2]
```diff
@@ -230,6 +232,18 @@ export default {
...
+            .catch((error) => console.log(error));
```
Comment (Vladislav94) [inline src/api/thegraph.api.js:238]: please move callback to the newline, not need wrap argument into brackets

File: src/api/thegraph.api.js [hunk 3]
```diff
@@ -230,6 +232,18 @@ export default {
...
+    },
+
+    async getErc1155ListingsBySeller(seller) {
+        return await this.getData(erc1155ListingsBySeller(seller))
+            .then(response => response.data.erc1155Listings)
```
Comment (Vladislav94) [inline src/api/thegraph.api.js:243]: please move callback to the newline

File: src/api/thegraph.api.js [hunk 4]
```diff
@@ -230,6 +232,18 @@ export default {
...
+            .catch((error) => console.log(error));
```
Comment (Vladislav94) [inline src/api/thegraph.api.js:244]: please move callback to the newline, not need wrap argument into brackets

File: src/pages/Listings/Listings.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+    const [portals, setPortals] = useState([]);
+    const [tickets, setTickets] = useState([]);
+    const [consumables, setConsumables] = useState([]);
+
+    useEffect(() => {
+        const currentAddress = params.address ? params.address : activeAddress;
+
+        if (currentAddress) {
+            getListings(currentAddress);
+
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:54]: extra whitespace

File: src/pages/Listings/Listings.js [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+    const getListings = (address) => {
+        setIsListingsLoading(true);
+
+        Promise.all([
+            thegraph.getErc721ListingsBySeller(address),
+            thegraph.getErc1155ListingsBySeller(address)
+        ]).then(([erc721Listings, erc1155Listings]) => {
+            const isListingsEmpty = erc721Listings.length === 0 && erc1155Listings.length === 0;
+            setIsListingsEmpty(isListingsEmpty);
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:70]: add whitespace after const

File: src/pages/Listings/Listings.js [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+            if (isListingsEmpty) {
+                setEmptyListings();
+            } else {
+                handleSetErc721Listings(erc721Listings);
+                handleSetErc1155Listings(erc1155Listings);
+            }
+        }).catch(error => {
+            console.log(error);
+
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:80]: extra whitespace

File: src/pages/Listings/Listings.js [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+            setEmptyListings();
+        }).finally(() =>setIsListingsLoading(false));
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:82]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+        const listedParcels = listings
+            .filter(listing => listing.category === ListingCategories.Fourth)
+            .map(listing => ({
+                ...listing.parcel,
+                priceInWei: listing.priceInWei,
+                baazaarId: listing.id
+            }));
+
+        const listedPortals = listings
+            .filter(listing => listing.category === ListingCategories.Zero || listing.category === ListingCategories.Second)
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:108]: please move callback to the newline

File: src/api/common/queries.js [hunk 10]
```diff
@@ -226,7 +311,7 @@ export const listedParcelsQuery = (skip, orderDir, size) => {
             skip: ${skip},
             orderDirection: ${orderDir},
             orderBy: timeCreated,
-            
+
```
Comment (Vladislav94) [inline src/api/common/queries.js:314]: extra whitespace

File: src/pages/Listings/Listings.js [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+        setParcels([]);
+        setPortals([]);
+        setWearables([]);
+        setTickets([]);
+        setConsumables([]);
+    }
+
+    const handleSetErc721Listings = (listings) => {
+        const listedGotchis = listings
+            .filter(listing => listing.category === ListingCategories.Third)
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:96]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+            .map(listing => listing.gotchi);
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:97]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+
+        const listedParcels = listings
+            .filter(listing => listing.category === ListingCategories.Fourth)
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:100]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 14]
```diff
@@ -0,0 +1,280 @@
...
+            }));
+
+        setGotchis(listedGotchis);
+        setParcels(listedParcels);
+        setPortals(listedPortals);
+    }
+
+    const handleSetErc1155Listings = (listings) => {
+        const wearables = listings
+            .filter(listing => listing.category === ListingCategories.Zero)
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:125]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 15]
```diff
@@ -0,0 +1,280 @@
...
+            .map(listing => ({
+                ...mapWearableAndTicket(listing)
+            }));
+
+        const tickets = listings
+            .filter(listing => listing.category === ListingCategories.Third)
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:131]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 16]
```diff
@@ -0,0 +1,280 @@
...
+            .map(listing => ({
+                ...mapWearableAndTicket(listing),
+                erc1155TypeId: listing.erc1155TypeId,
+            }));
+
+        const consumables = listings
+            .filter(listing => listing.category === ListingCategories.Second)
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:138]: please move callback to the newline

File: src/pages/Listings/Listings.js [hunk 17]
```diff
@@ -0,0 +1,280 @@
...
+            listing: listing.id,
+            rarity: itemUtils.getItemRarityById(listing.erc1155TypeId),
+            rarityId: listing.rarityLevel,
+            priceInWei: listing.priceInWei,
+            price: ethersApi.fromWei(listing.priceInWei)
+        }
+    }
+
+    if (isListingsLoading) {
+        return <div className={classes.loaderBox}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:165]: please wrap into brackets

File: src/pages/Listings/Listings.js [hunk 18]
```diff
@@ -0,0 +1,280 @@
...
+
+    return (
+        <div className={classes.container}>
+            {
+                Boolean(gotchis.length) && <ListingTitle src={gotchiPlaceholder} alt='gotchi' title='Gotchis' />
+            }
+
+            <div className={classes.list}>
+                {
+                    gotchis.map((gotchi, i)=>{
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:180]: add spaces before and after "=>"

File: src/pages/Listings/Listings.js [hunk 19]
```diff
@@ -0,0 +1,280 @@
...
+                        return <div className={classes.listItem} key={i}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:181]: please wrap into brackets

File: src/pages/Listings/Listings.js [hunk 20]
```diff
@@ -0,0 +1,280 @@
...
+
+
+            {
+                Boolean(wearables.length) && <ListingTitle src={warehousePlaceholder} alt='wearable' title='Wearables' />
+            }
+
+            <div className={classes.list}>
+                {
+                    wearables.map(wearable => {
+                        return <div className={classes.listItem} key={wearable.listing}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:215]: please wrap into brackets

File: src/pages/Listings/Listings.js [hunk 21]
```diff
@@ -0,0 +1,280 @@
...
+            </div>
+
+            {
+                Boolean(parcels.length) && <ListingTitle src={realmPlaceholder} alt='parcel' title='Parcels' />
+            }
+
+            <div className={classes.list}>
+                {
+                    parcels.map(parcel => {
+                        return <div className={classes.listItem} key={parcel.parcelId}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:229]: please wrap into brackets

File: src/pages/Listings/Listings.js [hunk 22]
```diff
@@ -0,0 +1,280 @@
...
+
+            {
+                Boolean(portals.length) && <ListingTitle src={portalPlaceholder} alt='portal' title='Portals' />
+            }
+
+
+            <div className={classes.list}>
+                {
+                    portals.map(portal => {
+                        return <div className={classes.listItem} key={portal.tokenId}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:244]: please wrap into brackets

File: src/pages/Listings/Listings.js [hunk 23]
```diff
@@ -0,0 +1,280 @@
...
+            </div>
+
+            {
+                Boolean(tickets.length) && <ListingTitle src={ticketsPlaceholder} alt='ticket' title='Tickets' />
+            }
+
+            <div className={classes.list}>
+                {
+                    tickets.map(ticket => {
+                        return <div className={classes.listItem} key={ticket.listing}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:258]: please wrap into brackets

File: src/pages/Listings/Listings.js [hunk 24]
```diff
@@ -0,0 +1,280 @@
...
+            </div>
+
+            {
+                Boolean(consumables.length) && <ListingTitle src={consumablePlaceholder} alt='consumable' title='Consumables' />
+            }
+
+            <div className={classes.list}>
+                {
+                    consumables.map(consumable => {
+                        return <div className={classes.listItem} key={consumable.listing}>
```
Comment (Vladislav94) [inline src/pages/Listings/Listings.js:272]: please wrap into brackets

File: src/pages/Listings/Listings.js (hunk 5 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:54]: it is logically separated

File: src/pages/Listings/Listings.js (hunk 7 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:80]: logically separated

File: src/api/thegraph.api.js (hunk 1 above)
Comment (dmutrokot) [reply src/api/thegraph.api.js:237]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/api/thegraph.api.js (hunk 3 above)
Comment (dmutrokot) [reply src/api/thegraph.api.js:243]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/pages/Listings/Listings.js (hunk 9 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:108]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/pages/Listings/Listings.js (hunk 13 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:100]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/pages/Listings/Listings.js (hunk 12 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:97]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/pages/Listings/Listings.js (hunk 14 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:125]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/pages/Listings/Listings.js (hunk 15 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:131]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/pages/Listings/Listings.js (hunk 16 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js:138]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/components/Items/ERC1155/ERC1155.js [hunk 25]
```diff
@@ -44,12 +44,17 @@ export default function ERC1155({ children, item }) {
...
-                setCurrent(response);
-            }
-        });
+        if (item.listing) {
+            setCurrent(item.listing);
+        } else {
+            // current
+            thegraph.getErc1155Price(item.id, false, item.category, 'priceInWei', 'asc').then((response) => {
+                if (!controller.signal.aborted) {
+                    console.log('current', response)
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Items/ERC1155/ERC1155.js]: removed `console.log`

File: src/api/thegraph.api.js (hunk 4 above)
Comment (dmutrokot) [reply src/api/thegraph.api.js]: this is pretty small callback, do not see a reason to do so, it's clear and readable. Wrapping removed

File: src/pages/Listings/Listings.js (hunk 20 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: return removed, no need to wrap

File: src/pages/Listings/Listings.js (hunk 22 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: return removed, no need to wrap

File: src/pages/Listings/Listings.js (hunk 24 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: return removed, no need to wrap

File: src/pages/Listings/Listings.js (hunk 21 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: return removed, no need to wrap

File: src/pages/Listings/Listings.js (hunk 23 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: return removed, no need to wrap

File: src/api/common/queries.js (hunk 10 above)
Comment (dmutrokot) [reply src/api/common/queries.js]: removed, thanks

File: src/pages/Listings/Listings.js (hunk 17 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: wrapped, thanks

File: src/pages/Listings/Listings.js (hunk 18 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: fixed, thanks

File: src/pages/Listings/Listings.js (hunk 19 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: return removed, no need to wrap

File: src/pages/Listings/Listings.js (hunk 8 above)
Comment (dmutrokot) [reply src/pages/Listings/Listings.js]: this is pretty small callback, do not see a reason to do so, it's clear and readable

File: src/api/thegraph.api.js (hunk 2 above)
Comment (dmutrokot) [reply src/api/thegraph.api.js]: this is pretty small callback, do not see a reason to do so, it's clear and readable. Wrapping removed

File: src/pages/Client/styles.js [hunk 26]
```diff
@@ -242,6 +242,31 @@ const loadRewardsStyles = makeStyles(theme => ({
 }));
 
 const parcelSinglePage = makeStyles(theme => ({
+
```
Comment (Vladislav94) [inline src/pages/Client/styles.js:245]: extra whitespace

Comment (dmutrokot) [reply src/pages/Client/styles.js]: fixed, thanks

File: src/components/Items/Parcel/Parcel.js [hunk 27]
```diff
@@ -36,52 +27,10 @@ export default function Parcel({ parcel, isBaazaarCard }) {
...
-            if (!controller.signal.aborted) {
-                setCurrent(response);
-            }
-        });
-
-        return () => controller?.abort(); // cleanup on destroy
-    }, [parcel]);
-
     return (
         <div className={classNames(classes.item, size, classes.parcelCard)}>
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Items/Parcel/Parcel.js:31]: fixed, thanks

**Review by Vladislav94 — APPROVED** [review]

**Pull request merged**